modules/imgproc/src/hough.cpp

```diff
@@ -180,7 +180,11 @@ HoughLinesStandard( InputArray src, OutputArray lines, int type,
                 {
                     int r = cvRound( j * tabCos[n] + i * tabSin[n] );
                     r += (numrho - 1) / 2;
-                    accum[(n+1) * (numrho+2) + r+1]++;
+                    if (use_edgeval) {
```

Comment:
`if` in inner loops is slow.

Reply:
I moved if(use_edgeval) to outside of for loop.
But I still remain to use this bool; use_edgeval to avoid code duplication, instead of defining another Hough function.